pkg/storage/gcs/gcs.go

```diff
@@ -0,0 +1,109 @@
+package gcs
+
+import (
+	"reflect"
+
+	corev1 "k8s.io/api/core/v1"
+
+	operatorapi "github.com/openshift/api/operator/v1"
+	imageregistryv1 "github.com/openshift/cluster-image-registry-operator/pkg/apis/imageregistry/v1"
+	regopclient "github.com/openshift/cluster-image-registry-operator/pkg/client"
+	"github.com/openshift/cluster-image-registry-operator/pkg/clusterconfig"
+	"github.com/openshift/cluster-image-registry-operator/pkg/storage/util"
+)
+
+type driver struct {
+	Config  *imageregistryv1.ImageRegistryConfigStorageGCS
+	Listers *regopclient.Listers
+}
+
+func NewDriver(c *imageregistryv1.ImageRegistryConfigStorageGCS, listers *regopclient.Listers) *driver {
+	return &driver{
+		Config:  c,
+		Listers: listers,
+	}
+}
+
+func (d *driver) Secrets() (map[string]string, error) {
+	cfg, err := clusterconfig.GetGCSConfig(d.Listers)
+	if err != nil {
+		return nil, err
+	}
+
+	return map[string]string{
+		"STORAGE_GCS_KEYFILE": cfg.Storage.GCS.KeyfileData,
+	}, nil
+}
+
+func (d *driver) ConfigEnv() (envs []corev1.EnvVar, err error) {
+	envs = append(envs,
+		corev1.EnvVar{Name: "REGISTRY_STORAGE", Value: "gcs"},
+		corev1.EnvVar{Name: "REGISTRY_STORAGE_GCS_BUCKET", Value: d.Config.Bucket},
+		corev1.EnvVar{Name: "REGISTRY_STORAGE_GCS_KEYFILE", Value: "/gcs/keyfile"},
+	)
+	return
+}
+
+func (d *driver) Volumes() ([]corev1.Volume, []corev1.VolumeMount, error) {
+	vol := corev1.Volume{
+		Name: "registry-storage-keyfile",
+		VolumeSource: corev1.VolumeSource{
+			Projected: &corev1.ProjectedVolumeSource{
```

Comment:
```
Secret: &corev1.SecretVolumeSource{
```

There is no need to use a projected volume source.

Reply:
Updated in current push